### PR TITLE
API Deprecate renamed validator

### PR DESCRIPTION
**Pull request by emteknetnz**
Issue https://github.com/silverstripe/silverstripe-framework/issues/10908

### CODE REVIEW

File: src/Forms/AWRequiredFields.php [hunk 1]
```diff
@@ -17,6 +18,16 @@ class AWRequiredFields extends RequiredFields
...
     protected static $caller;
 
+    /**
+     * @deprecated 5.4.0 Will be renamed to Symbiote\AdvancedWorkflow\Forms\AWRequiredFieldsValidator
+     */
+    public function __construct()
+    {
+        $message = 'Will be renamed to Symbiote\\AdvancedWorkflow\\Forms\\AWRequiredFieldsValidator';
+        Deprecation::noticeWithNoReplacment('5.4.0', $message, Deprecation::SCOPE_CLASS);
+        parent::__construct(...func_get_args());
```
Comment (emteknetnz) [inline src/Forms/AWRequiredFields.php:27]: RequiredFields constructor uses func_get_args() https://github.com/silverstripe/silverstripe-framework/blob/5/src/Forms/RequiredFields.php#L31

File: src/Forms/AWRequiredFields.php [hunk 2]
```diff
@@ -11,12 +12,21 @@
...
+ * @deprecated 5.4.0 Will be renamed to Symbiote\AdvancedWorkflow\Forms\AWRequiredFieldsValidator
  */
 class AWRequiredFields extends RequiredFields
 {
     protected $data = array();
     protected static $caller;
 
+    public function __construct()
+    {
+        $message = 'Will be renamed to Symbiote\\AdvancedWorkflow\\Forms\\AWRequiredFieldsValidator';
```
Comment (ChloeSartorelli) [inline src/Forms/AWRequiredFields.php:25]: May need to update depending on https://github.com/silverstripe/silverstripe-userforms/pull/1354#discussion_r1870393853 regarding namespace vs just class name

Comment (emteknetnz) [reply src/Forms/AWRequiredFields.php:25]: Will leave as is

**Review by GuySartorelli — APPROVED** [review]
LGTM

**Pull request merged**